robocopkaka/bootcamp-29-project
# #158284357 Make links dynamic on logout

#### What does this PR do?
* change navbar links on logout
* show user's name on the navbar when they are signed in
#### Description of Task to be completed?
* delete `registerActions.js`
* move actions for registering to `sessionActions.js`
* delete the `registerApi` file
* move the api method for registering users to the `sessionApi` class
* delete the `registerReducer` file
* move all reducers for registering to the `sessionReducer` filer
* delete the register object in initial state
* modify tests for register actions
* move tests for register reducers to tests for session reducers
* remove the register key from the root reducer
* modify the Header component to show the user's name when logged in
* modify tests for the `Signup` component
* modify tests for the `Header` component
* modify the `Signup` component to use sessionActions
#### How should this be manually tested?
* clone this repo
* cd into the project folder
* run `npm install` to install all dependecies
* run `npm run start:all` to start up the application
* go to `localhost:8000` on your browser
* click on `Signup`
* enter your details
* click on `Signup`
* click on `Login`
* enter your details
* click on `Login`
* you should see your name at the top right corner of the page
* click on the your name
* click on `Logout`
* you should see `Login` at the top right corner of the page
#### Any background context you want to provide?
users had to refresh the page after logging out to see a `Login` link
#### What are the relevant pivotal tracker stories?(if applicable)
#158284357
